### ISSUE TITLE
Bitwarden 2024.8.0 regression causes sfcu.org to render blank in Firefox (related to the extension's injected <script src="...fido2-page-script.js">` element interfering with the site's page load logic)

### ISSUE THREAD

**Issue by dholbert** [bitwarden/clients#10798]
### Steps To Reproduce

1. Install BitWarden extension in Firefox. (No need to sign in.)
2. Visit https://www.sfcu.org/

### Expected Result

Page should render.

### Actual Result

Page does not render; it's just blank.

### Screenshots or Videos

![image](https://github.com/user-attachments/assets/df1f1c35-df5d-4baa-9490-be01627cbc31)

### Additional Context

The page remains blank because it never reaches its JS that would adds the `delayfade` class to its `body` element.  (If I add that class manually in devtools, then the page renders properly.)

This seems to be a case where the site seems to have a fragile dependency on how its `script` resources load, and BitWarden happens to insert its own `script` element into the page:
```html
<script src="moz-extension://c9b14213-ffcd-40ee-9488-bf198b862aee/content/fido2-page-script.js" type="text/plain">
```
...and that script element ends up as the final member in the `mutationNodes` array here, in each call to this `dequeueNonAsyncScripts` function in the site's cookiebot logic:
```JavaScript
  function dequeueNonAsyncScripts(cookieConsent, mutationNodes) {
    if (mutationNodes.length > 0) {
```
https://consent.cookiebot.com/uc.js

If I explicitly remove it when we reach this function (by calling `mutationNodes.pop()`), then everything works fine.
[EDIT: See also the screencast that I linked in a comment below where I go into a bit more detail on this.]

### Operating System

Linux

### Operating System Version

22.04

### Web Browser

Firefox

### Browser Version

131.0a1 (2024-08-29) Nightly

### Build Version

Version: 2024.8.0

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by dholbert** [comment]
This seems to be a regression in the version of the BitWarden add-on that was just released.

If I install the previous [version 2024.7.1](https://addons.mozilla.org/firefox/downloads/file/4326285/bitwarden_password_manager-2024.7.1.xpi) from the [addons.mozilla.org version history page](https://addons.mozilla.org/en-US/firefox/addon/bitwarden-password-manager/versions/), then https://www.sfcu.org/ loads just fine.

Whereas with the latest version, [version 2024.8.0](https://addons.mozilla.org/firefox/downloads/file/4340992/bitwarden_password_manager-2024.8.0.xpi), SFCU triggers this issue and ends up blank.

**Comment by dholbert** [comment]
(ack, it seems I forgot to type a summary; I just fixed that now, sorry)

As a bit of a shot-in-the-dark -- looking at recent commit history, I see @jprusik has some commits mentioning fido in the past weeks (possibly relevant to the injected `fido2-page-script.js` script that's involved with causing trouble here) - @jprusik, do you possibly have ideas about what might have caused this regression or what might be going on here? (or know who else might be worth poking about this)

Thanks!

**Comment by dholbert** [comment]
Looking more at the commit history, I'd guess this is related to (maybe/likely regressed by) #10424 which was merged on August 7th.  That PR's first comment talks about the Fido2 page script (the script in question here) and stuff happening around page-load (the timeframe in question here).

CC @ cagonzalezcs

**Comment by dumbbell** [comment]
Hi!

I get a similar problem with https://robertsspaceindustries.com/ which doesn’t render properly. It uses the same cookiebot cookie manager as sfcu.org.

**Comment by dholbert** [comment]
Here's a screencast showing the issue, and showing how I can work around it by manually popping BitWarden's fido script from a JS array that cookiebot's `dequeueNonAsyncScripts` function processes:
[Screencast from 08-29-2024 10:29:13 AM.webm](https://github.com/user-attachments/assets/8aeaf4b9-6a98-438c-ad46-0bdc3dd542ef)

Rough walkthrough of the screencast:
* From t=0s to t=10s: I reload the page several times, showing that it's blank.
* At t=10s, I enable a JS debugger breakpoint in https://consent.cookiebot.com/uc.js (at line 1151 in the pretty-printed (`{ }` icon) source), in `dequeueNonAsyncScripts`.
* Then I reload the page, and when I hit that breakpoint, I inspect the final element of the `mutationNodes` array (at index `length-1`) that `dequeueNonAsyncScripts` is working with.  It's this `fido2-page-script.js` script from BitWarden.
* I intervene to `.pop()` that element from the array so that the function doesn't have to worry about it, and then I disable the breakpoint and resume execution.
* t=32s: The page completes loading and renders just fine.
 
(When the page loads just fine, it happens via a call to `endJQueryHold` in cookiebot's source, if you're curious - you can see that one of the disabled breakpoints in my screencast.  That function is what's resopnsible for adding the `delayfade` class to the `body` element, which makes it show up; and that function never gets called unless I remove bitwarden's script from this `mutationNodes` array.)

**Comment by dholbert** [comment]
> Hi!
> 
> I get a similar problem with https://robertsspaceindustries.com/ which doesn’t render properly. It uses the same cookiebot cookie manager as sfcu.org.

Thanks - yeah, I can confirm the issue at https://robertsspaceindustries.com -- here's a side-by-side comparison, with BitWarden disabled on the left (fine) vs. enabled on the right (broken):
![image](https://github.com/user-attachments/assets/bec65b8a-82e9-4136-97be-5f9198d791f4)

As with SFCU, I can work around this using the same manual JS-debugger intervention that I described/screencasted in my previous comment.

**Comment by cagonzalezcs** [comment]
@dholbert 

Thank you for the incredibly detailed bug report. 

Hmm, so this is definitely frustrating on our side as well. [The fix you've linked above](https://github.com/bitwarden/clients/pull/10424) was mean to resolve an issue found within a series of web clients [that run into issues rendering](https://github.com/bitwarden/clients/issues/9618) when that `fido2-page-script.ts` file is loaded before DOM content has loaded entirely. 

The issue with our fix appears to be in relation to how https://consent.cookiebot.com/uc.js is handling mutations... within the content script that we are using to inject this `fido2-page-script.ts` file, we are triggering a cleanup of our script on `load` of the script itself. 

The problem with that is that coookiebot isn't expecting us to remove the element from the DOM, and it's expecting to have a reference to modify when it is processing mutations on the page... 

We have a couple of silver linings to this situation though...

The first is that this work is behind a feature flag. I'm speaking with my manager on whether it makes sense to revert our fix given the potential scope of issues introduced to websites that use https://consent.cookiebot.com/uc.js. I'll provide an update as soon as I can on whether we're turning the flag that introduces this fix/issue off (please keep in mind that fixing your website's issue by turning the flag off will break a number of webmail clients, so we will be considering those concerns on our side). 

The second silver lining is that I'm not seeing this problem appear in Firefox v129.0.2, which is the latest release... I wonder why that would be the case, as the problem likely should appear in that version as well as the nightly build.

Can you confirm whether the problem appears on your side within the prod release of Firefox?

**Comment by cagonzalezcs** [comment]
Hmm... it seems that an update was released for https://consent.cookiebot.com/uc.js yesterday that resolves the issue experienced here.

From the first line of the script:
```// 2.74.0 - 2024-08-29T10:53:00.105Z```

Can you confirm @dholbert whether that is the case? I'm seeing on my side that the page is loading as expected now that they have updated their script. 

I've got a PR in review that would further fix this issue from our side as well - https://github.com/bitwarden/clients/pull/10830

**Comment by dholbert** [comment]
I'm still seeing the issue locally (and I'm seeing that I've got that version of the script).

Screenshot of a fresh Firefox profile, running latest Firefox Nightly with latest BitWarden extension, loading sfcu.org (blank), with debugger open to show `uc.js` script which starts with `// 2.74.0 - 2024-08-29T10:53:00.105Z`:
![image](https://github.com/user-attachments/assets/28a8ff09-4204-4f12-aeb4-ea9174107098)

(And I confirmed that sfcu.org loads fine if I disable Bitwarden.)

**Comment by dumbbell** [comment]
I tried both https://www.sfcu.org/ and https://robertsspaceindustries.com/, and none of them rendered correctly unfortunately. I double-checked the new `uc.js` was fetched and the `// 2.74.0 - 2024-08-29T10:53:00.105Z` is present.

**Comment by dholbert** [comment]
( @cagonzalezcs since it sounds like you're having little/mixed success reproducing on your end yesterday and/or today -- I want to be sure you saw [my earlier comment](https://github.com/bitwarden/clients/issues/10798#issuecomment-2319964795) with some possible explanations.  If you test in a regular Firefox window with default tracking protection (not strict) and no other content-blocking extensions (just Bitwarden), I *think* this should reproduce 100% reliably.

Easiest way is in a fresh Firefox profile -- if it's helpful, you can visit `about:profiles` in Firefox which has a button to generate a new Firefox profile for testing (though that new profile automatically becomes your default, which you probably don't want -- so after you've created one there, take care to click "set as default profile" on the listing there for your 'real' profile).

**Comment by AlexHaaniWink** [comment]
@cagonzalezcs You marked this closed 2 weeks ago. I see that you added https://github.com/bitwarden/clients/pull/10830 3 weeks ago.

Am I correct to **not** see that PR being part of the 2024.9.0 release (the version I'm running in my Linux Firefox install)? At least I don't see it in [the v2024.8.0 -> v2024.9.0 compare](https://github.com/bitwarden/clients/compare/web-v2024.8.0...browser-v2024.9.0). But maybe I'm comparing this wrong.

This issue makes working on better Cookiebot support for our websites a bit tricky.

**Comment by dholbert** [comment]
Following up on @AlexHaaniWink's note -- I'm running version add-on version 2024.9.0 (released a few days ago) in latest Firefox Nightly and I can still reproduce this problem, for what it's worth.  https://www.sfcu.org/ renders blank for me.

**Comment by cagonzalezcs** [comment]
@AlexHaaniWink @dholbert 

The fix for this issue is being released in `v2024.9.1`, which should be going out this week. 

For the immediate, you can fix this problem by turning off the Passkeys feature in the extension by going to `Settings > Notifications > Ask to save and use passkeys` and checking that setting off.

**Comment by AlexHaaniWink** [comment]
@cagonzalezcs Thank you very much!
